soweego/importer/imdb_dump_extractor.py

```diff
@@ -0,0 +1,390 @@
+        LOGGER.debug("Path to movie info dump: %s", movies_file_path)
+        LOGGER.debug("Path to person info dump: %s", person_file_path)
+
+        start = datetime.datetime.now()
+
+        tables = [
+            imdb_entity.ImdbActorEntity,
+            imdb_entity.ImdbDirectorEntity,
+            imdb_entity.ImdbMovieEntity,
+            imdb_entity.ImdbMusicianEntity,
+            imdb_entity.ImdbProducerEntity,
+            imdb_entity.ImdbWriterEntity,
+            imdb_entity.ImdbPersonMovieRelationship,
+        ]
+
+        db_manager = DBManager()
+        LOGGER.info("Connected to database: %s", db_manager.get_engine().url)
+
+        db_manager.drop(tables)
+        db_manager.create(tables)
+
+        LOGGER.info("SQL tables dropped and re-created: %s",
+                    [table.__tablename__ for table in tables])
+
+        LOGGER.info("Starting to import movies from imdb dump")
+
+        # Here we open the movie dump file, and add everything to the DB
+        with gzip.open(movies_file_path, "rt") as mdump:
+            reader = csv.DictReader(mdump, delimiter="\t")
+
+            # count number of rows for TQDM (so we can display how)
+            # much is missing to complete the process. Then go back
+            # to the start of the file with `.seek(0)`
+            n_rows = sum(1 for line in mdump)
+            mdump.seek(0)
+
+            LOGGER.debug("Movies dump has %d entries", n_rows)
+
+            session = db_manager.new_session()
+
+            for movie_info in tqdm(reader, total=n_rows):
+                self._normalize_null(movie_info)
+
+                # create the movie SQLAlchemy entity and populate it
+                movie_entity = imdb_entity.ImdbMovieEntity()
+                movie_entity.catalog_id = movie_info.get("tconst")
+                movie_entity.title_type = movie_info.get("titleType")
+                movie_entity.primary_title = movie_info.get("primaryTitle")
+                movie_entity.original_title = movie_info.get("originalTitle")
+                movie_entity.is_adult = True if movie_info.get(
+                    "isAdult") == "1" else False
+                movie_entity.start_year = movie_info.get("startYear")
+                movie_entity.end_year = movie_info.get("endYear")
+                movie_entity.runtime_minutes = movie_info.get("runtimeMinutes")
+
+                if movie_info.get("genres"):  # if movie has a genre specified
+                    movie_entity.genres = movie_info.get("genres").replace(",", " ")
+
+                session.add(movie_entity)
+
+                # every `_sqlalchemy_commit_every` loops we commit the session to
+                # the DB. This is more efficient than commiting every loop, and
+                # is not so hard on the memory requirements as would be
+                # adding everything to session and commiting once the for loop
+                # is done
+                if self.n_movies % self._sqlalchemy_commit_every == 0:
```

Comment:
Great idea, this should be implemented in MusicBrainz and Discogs as well.
Are there specific reasons why you chose `700` entities as the value?

Reply:
New issues to updates MusicBrainz and Discogs importers have been added to the issue tracker.

And not really, I just tried different values and a batch size of `700` seemed to be the one that performed best on my PC. Do you think we should calculate this value in some other way?

Reply:
> a batch size of `700` seemed to be the one that performed best on my PC. Do you think we should calculate this value in some other way?

It's most probably fine, but we should think of the prodcution server, which has quite constrained settings. I'm gonna give you access to our VPS machine, so you can try it and check whether the value you chose is fine.